torch/utils/dlpack.py

```diff
@@ -104,24 +106,34 @@ def from_dlpack(ext_tensor: Any) -> 'torch.Tensor':
 
     """
     if hasattr(ext_tensor, '__dlpack__'):
+        kwargs: dict[str, Any] = {}
+        kwargs["max_version"] = (1, 0)
+
         device = ext_tensor.__dlpack_device__()
         # device is either CUDA or ROCm, we need to pass the current
         # stream
-        if device[0] in (DLDeviceType.kDLGPU, DLDeviceType.kDLROCM):
+        if device[0] in (DLDeviceType.kDLCUDA, DLDeviceType.kDLROCM):
             stream = torch.cuda.current_stream(f'cuda:{device[1]}')
             # cuda_stream is the pointer to the stream and it is a public
             # attribute, but it is not documented
             # The array API specify that the default legacy stream must be passed
             # with a value of 1 for CUDA
             # https://data-apis.org/array-api/latest/API_specification/array_object.html?dlpack-self-stream-none#dlpack-self-stream-none
-            is_cuda = device[0] == DLDeviceType.kDLGPU
+            is_cuda = device[0] == DLDeviceType.kDLCUDA
             # Since pytorch is not using PTDS by default, lets directly pass
             # the legacy stream
             stream_ptr = 1 if is_cuda and stream.cuda_stream == 0 else stream.cuda_stream
-            dlpack = ext_tensor.__dlpack__(stream=stream_ptr)
-        else:
-            dlpack = ext_tensor.__dlpack__()
+            kwargs["stream"] = stream_ptr
+
+        try:
+            # Try running __dlpack__ while specifying `max_version` argument.
+            dlpack = ext_tensor.__dlpack__(**kwargs)
+        except TypeError:
```

Comment:
According to the standard, it says BufferError should be raised if you don't get a good max_version. Should we handle that here?